generate-release/src/github_client.rs

```diff
@@ -305,8 +336,83 @@ query {{
         Ok(contributors)
     }
 
-    pub fn get_commit(&self, git_ref: &str) -> anyhow::Result<GithubCommitResponse> {
-        let request = self.get(&format!("commits/{git_ref}"));
+    /// Gets the data for a specific commit on the provided `bevyengine` repo.
+    pub fn get_commit(&self, git_ref: &str, repo: &str) -> anyhow::Result<GithubCommitResponse> {
+        let request = self.get(&format!("commits/{git_ref}"), repo);
         Ok(request.call()?.into_json()?)
     }
+
+    /// Opens a new issue on the `bevyengine/bevy-website` repo.
+    ///
+    /// See [the Github API documentation](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#create-an-issue) for more information.
+    #[allow(clippy::result_large_err)]
+    pub fn open_issue(
+        &self,
+        repo: &str,
+        issue_title: &str,
+        issue_body: &str,
+        labels: Vec<&str>,
+    ) -> Result<(), IssueError> {
+        let response = self
+            .agent
+            .post(&format!(
+                "https://api.github.com/repos/bevyengine/{repo}/issues"
+            ))
+            .set("Authorization", &format!("Bearer {}", self.token))
+            .set("Accept", "application/vnd.github+json")
+            .set("X-GitHub-Api-Version", "2022-11-28")
+            .send_json(ureq::json!({
+                "title": issue_title,
+                "body": issue_body,
+                // TODO: add the milestone. Tracked in https://github.com/bevyengine/bevy-website/issues/1269
+                // Note that this must be provided as an integer, so we'll have to look up the milestone ID.
+                "labels": labels,
+            }))?;
+
+        // Make sure the issue was created successfully
+        if response.status() != 201 {
+            Err(IssueError::FailedToCreateIssue(response))
+        } else {
+            Ok(())
+        }
+    }
+}
+
+/// An issue that occurred while opening an issue on Github.
+#[derive(Debug)]
+pub enum IssueError {
+    Ureq(ureq::Error),
+    FailedToCreateIssue(Response),
+}
+
+impl From<ureq::Error> for IssueError {
+    fn from(err: ureq::Error) -> Self {
+        IssueError::Ureq(err)
+    }
+}
+
+/// The status of an issue or PR on Github.
+#[derive(Debug, Clone, Copy)]
+#[allow(dead_code)]
+pub enum IssueState {
+    Open,
+    Closed,
+    Merged,
+    All,
+}
```

Comment:
```suggestion
#[derive(Debug, Clone, Copy)]
pub enum IssueState {
    #[allow(dead_code)]
    Open,
    #[allow(dead_code)]
    Closed,
    Merged,
    All,
}
```

This makes it still lint a warning if `IssueState` is never used at all, but ignore any warnings about `Open` and `Closed` specifically.